lib/active_fedora/noid/synchronized_minter.rb

```diff
@@ -38,16 +38,20 @@ def state_for(io_object)
         { template: template }
       end
 
+      def file_opts
+        { encoding: Encoding::ASCII_8BIT }
+      end
+
       def next_id
         id = ''
-        ::File.open(statefile, ::File::RDWR|::File::CREAT, 0644) do |f|
+        ::File.open(statefile, ::File::RDWR|::File::CREAT, 0644, file_opts) do |f|
```

Comment:
Setting this to `::File.open(statefile, 'r+b', 0644) do |f| ...` is how I'd do this. You should be able to avoid lines 54 & the file_opts method then.